portal-impl/src/com/liferay/portlet/internal/LiferayPortletURLPrivilegedAction.java

```diff
@@ -139,7 +139,7 @@ public LiferayPortletURL run() {
 			}
 			else {
 				portletURL = PortletURLFactoryUtil.create(
-					_portletRequest, _portletName, plid, _lifecycle, _copy);
+					_portletRequest, portletId, plid, _lifecycle, _copy);
```

Comment:
@adolfopa I just realized that making this change we are calling 2 times to 
`portletURL = PortletURLFactoryUtil.create(_portletRequest, _portlet, plid, _lifecycle, _copy)`; (line 137)
because the portlet id is obtained from `_portlet`

So I think that this solution is not working